### PR TITLE
fix old password match

### LINKED ISSUE #12584: Password Change Not Validating Old Password
In the user profile, system is not validating the old password entered by the user when they attempt to change their password. This means that a user can change their password to any value, even if it is the same as their old password.
Steps to Reproduce:
Log in to the user profile system.
Click on the "Change Password" link.
Enter your old password.
Enter a new password.
Enter the new password again to confirm.
Click on the "Update Password" button.
Expected Results:
The system should validate the old password and only allow the user to change their password if the old password is correct.

### PR DESCRIPTION
**Pull request by harsh-vador**
### Describe your changes:

Fixes #12584 

I worked on fixing the old password validation not working while updating new password
For User side:

https://github.com/open-metadata/OpenMetadata/assets/58542468/3830237c-0184-4921-a3e3-482fa0820456

For admin side:

https://github.com/open-metadata/OpenMetadata/assets/58542468/03d7c570-71ec-43df-9a83-0a70bca803d4

#
### Type of change:

- [x] Bug fix

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Users/UsersProfile/UserProfileDetails/UserProfileDetails.component.tsx [hunk 1]
```diff
@@ -171,27 +171,38 @@ const UserProfileDetails = ({
...
+        username: '',
+        requestType: RequestType.User,
+      };
+
+      if (isLoggedInUser) {
+        newData.username = userData.name;
+        newData.requestType = RequestType.Self;
+      } else {
+        newData.username = userData.name;
+      }
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/UsersProfile/UserProfileDetails/UserProfileDetails.component.tsx:187]:
```suggestion
      const newData = {
        username: userData.name,
        requestType: isLoggedInUser ? RequestType.Self : RequestType.User,
      };

```

File: openmetadata-ui/src/main/resources/ui/src/components/Users/UsersProfile/UserProfileDetails/UserProfileDetails.component.tsx [hunk 2]
```diff
@@ -171,27 +171,38 @@ const UserProfileDetails = ({
...
+      await changePassword(sendData)
+        .then(() => {
+          showSuccessToast(
+            t('server.update-entity-success', { entity: t('label.password') })
+          );
+          setIsChangePassword(false);
+        })
+        .catch((error) => {
+          showErrorToast(error.response.data.message);
+        });
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/UsersProfile/UserProfileDetails/UserProfileDetails.component.tsx:201]: await and then can't be used to gather, We have async await syntax for async operation in the code

**Review by chirag-madlani — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/components/Users/UsersProfile/UserProfileDetails/UserProfileDetails.component.tsx (hunk 1 above)
Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/src/components/Users/UsersProfile/UserProfileDetails/UserProfileDetails.component.tsx]: addressed comments

File: openmetadata-ui/src/main/resources/ui/src/components/Users/UsersProfile/UserProfileDetails/UserProfileDetails.component.tsx [hunk 3]
```diff
@@ -171,27 +171,35 @@ const UserProfileDetails = ({
...
-    } catch (err) {
-      showErrorToast(err as AxiosError);
+
+      setIsChangePassword(false);
+    } catch (error) {
+      if (error instanceof AxiosError) {
+        showErrorToast(error.response?.data?.message || 'An error occurred');
+      } else {
+        showErrorToast('An error occurred');
+      }
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Users/UsersProfile/UserProfileDetails/UserProfileDetails.component.tsx:200]:
```suggestion
showErrorToast(error);
```

Comment (harsh-vador) [reply openmetadata-ui/src/main/resources/ui/src/components/Users/UsersProfile/UserProfileDetails/UserProfileDetails.component.tsx:200]: addressed

**Review by chirag-madlani — APPROVED** [review]

**Review by 07Himank — APPROVED** [review]

**Pull request merged**